test/e2e/framework/util.go

```diff
@@ -1246,33 +1257,63 @@ func WaitForPodNoLongerRunningInNamespace(c clientset.Interface, podName, namesp
 }
 
 func WaitTimeoutForPodNoLongerRunningInNamespace(c clientset.Interface, podName, namespace, resourceVersion string, timeout time.Duration) error {
-	w, err := c.Core().Pods(namespace).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: podName, ResourceVersion: resourceVersion}))
-	if err != nil {
-		return err
+	return wait.PollImmediate(10*time.Millisecond, timeout, podCompleted(c, podName, namespace))
```

Comment:
The watch can also have "timeout" set explicitly (this is what we are doing e.g. in reflector framework). That said, I don't think we are doing it in tests (though we should probably check it).

Anyway - I agree with @janetkuo that it can happen that watch is closed not due to timeout, but e.g. immediately due to "rv too old" error, or some other issue).

But sure, polling every 10ms might be too frequently. Maybe every 100ms or sth like that?